core/platform/search/gae_search_services.py

```diff
@@ -48,7 +48,7 @@ def add_documents_to_index(documents, index, retries=DEFAULT_NUM_RETRIES):
     """Adds a document to an index.
 
     Args:
-        documents: list. Each document should be a dictionary.
+        documents: list(dict(str, str)). Each document should be a dictionary.
```

Comment:
Are you certain that all values for each dict are strings? If not, use list(dict).

Reply:
Are you sure? From the tests, the value for "rank" seems to be an int.

Reply:
I'll change it list(dict) because the tests have cases of str -> str and str -> int. My logic was since the documents argument is processed using the function "_dict_to_search_document(d)" and that function already had defined the document dict(str, str) (from the previous code owner), I extrapolated the typing. Should I change the typing of _dict_to_search_document(d) too?

Reply:
Ah yep, please fix that too -- thanks!

Reply:
Done